dart/lib/src/event_processor/enricher/io_platform_memory.dart

```
@@ -0,0 +1,107 @@
+class PlatformMemory {
+  PlatformMemory(this.operatingSystem);
+
+  final String operatingSystem;
+
+  int? getTotalPhysicalMemory() {
+    switch (operatingSystem) {
+      case 'linux':
+        return _getLinuxMemInfoValue('MemTotal');
+      case 'windows':
+        return _getWindowsWmicValue('ComputerSystem', 'TotalPhysicalMemory');
+      default:
+        return null;
+    }
+  }
+
+  int? getFreePhysicalMemory() {
+    switch (operatingSystem) {
+      case 'linux':
+        return _getLinuxMemInfoValue('MemFree');
+      case 'windows':
+        return _getWindowsWmicValue('OS', 'FreePhysicalMemory');
+      default:
+        return null;
+    }
+  }
+
+  int? _getWindowsWmicValue(String section, String key) {
+    final os = _wmicGetValueAsMap(section, [key]);
+    final totalPhysicalMemoryValue = os?[key];
+    if (totalPhysicalMemoryValue == null) {
+      return null;
+    }
+    final size = int.tryParse(totalPhysicalMemoryValue);
+    if (size == null) {
+      return null;
+    }
+    return size;
+  }
+
+  int? _getLinuxMemInfoValue(String key) {
+    final meminfoList = _exec('cat', ['/proc/meminfo'])
+            ?.trim()
+            .replaceAll('\r\n', '\n')
+            .split('\n') ??
+        [];
+
+    final meminfoMap = _listToMap(meminfoList, ':');
+    final memsizeResults = meminfoMap[key]?.split(' ') ?? [];
+
+    if (memsizeResults.isEmpty) {
+      return null;
+    }
+    final memsizeResult = memsizeResults.first;
+
+    final memsize = int.tryParse(memsizeResult);
+    if (memsize == null) {
+      return null;
+    }
+    return memsize;
+  }
+
+  String? _exec(String executable, List<String> arguments,
+      {bool runInShell = false}) {
+    try {
+      final result =
+          Process.runSync(executable, arguments, runInShell: runInShell);
+      if (result.exitCode == 0) {
+        return result.stdout.toString();
+      }
+    } catch (e) {
+      //
```

maybe we could log here with `options.logger` so exceptions are not swallowed